kvdb/src/lib.rs

```diff
@@ -70,48 +62,36 @@ impl DBOp {
 	}
 }
 
-impl DBTransaction {
+impl<K, V> DBTransaction<K, V> {
 	/// Create new transaction.
-	pub fn new() -> DBTransaction {
+	pub fn new() -> Self {
 		DBTransaction::with_capacity(256)
 	}
 
 	/// Create new transaction with capacity.
-	pub fn with_capacity(cap: usize) -> DBTransaction {
+	pub fn with_capacity(cap: usize) -> Self {
 		DBTransaction {
 			ops: Vec::with_capacity(cap)
 		}
 	}
 
 	/// Insert a key-value pair in the transaction. Any existing value will be overwritten upon write.
-	pub fn put(&mut self, col: Option<u32>, key: &[u8], value: &[u8]) {
-		let mut ekey = ElasticArray32::new();
-		ekey.append_slice(key);
+	pub fn put<IK, IV>(&mut self, col: Option<u32>, key: IK, value: IV)
+	where IK: Into<K>,
+		  IV: Into<V>,
```

Comment:
There are some spaces. Could you please change them to tabs?

Reply:
done